pluginsdk/support/bundlepublisherutil/bputil.go

```diff
@@ -0,0 +1,257 @@
+	switch bundleFormat {
+	case BundleFormatUnset:
+		return "UNSET"
+	case SPIFFE:
+		return "spiffe"
+	case PEM:
+		return "pem"
+	case JWKS:
+		return "jwks"
+	default:
+		return fmt.Sprintf("UNKNOWN(%d)", int(bundleFormat))
+	}
+}
+
+// Bytes returns the bundle in the form of a slice of bytes in
+// the chosen format.
+func (b *Bundle) Bytes(format BundleFormat) ([]byte, error) {
+	if b.bundle == nil {
+		return nil, errors.New("missing bundle")
+	}
+
+	switch format {
+	case BundleFormatUnset:
+		return nil, errors.New("no format specified")
+	case JWKS:
+		if b.jwksBytes != nil {
+			return b.jwksBytes, nil
+		}
+		jwksBytes, err := b.toJWKS()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to jwks format: %w", err)
+		}
+		b.jwksBytes = jwksBytes
+		return jwksBytes, nil
+	case PEM:
+		if b.pemBytes != nil {
+			return b.pemBytes, nil
+		}
+		pemBytes, err := b.toPEM()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to pem format: %w", err)
+		}
+		b.pemBytes = pemBytes
+		return pemBytes, nil
+	case SPIFFE:
+		if b.spiffeBytes != nil {
+			return b.spiffeBytes, nil
+		}
+		spiffeBytes, err := b.toSPIFFEBundle()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to spiffe format: %w", err)
+		}
+		b.spiffeBytes = spiffeBytes
+		return spiffeBytes, nil
+	default:
+		return nil, fmt.Errorf("invalid format: %q", format)
+	}
+}
+
+// toJWKS converts to JWKS the current bundle.
+func (b *Bundle) toJWKS() ([]byte, error) {
+	var jwks jose.JSONWebKeySet
+
+	x509Authorities, jwtAuthorities, err := getAuthorities(b.bundle)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rootCA := range x509Authorities {
+		jwks.Keys = append(jwks.Keys, jose.JSONWebKey{
+			Key:          rootCA.PublicKey,
+			Certificates: []*x509.Certificate{rootCA},
+		})
+	}
+
+	for keyID, jwtSigningKey := range jwtAuthorities {
+		jwks.Keys = append(jwks.Keys, jose.JSONWebKey{
+			Key:   jwtSigningKey,
+			KeyID: keyID,
+		})
+	}
+
+	var out interface{} = jwks
+	return json.MarshalIndent(out, "", "    ")
+}
+
+// toPEM converts to PEM the current bundle.
+func (b *Bundle) toPEM() ([]byte, error) {
+	bundleData := new(bytes.Buffer)
+	for _, x509Authority := range b.bundle.X509Authorities {
+		if err := pem.Encode(bundleData, &pem.Block{
+			Type:  "CERTIFICATE",
+			Bytes: x509Authority.Asn1,
+		}); err != nil {
+			return nil, fmt.Errorf("could not perform PEM encoding: %w", err)
+		}
+	}
+
+	return bundleData.Bytes(), nil
+}
+
+// toSPIFFEBundle converts to a SPIFFE bundle the current bundle.
+func (b *Bundle) toSPIFFEBundle() ([]byte, error) {
+	sb, err := spiffeBundleFromPluginProto(b.bundle)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert bundle: %w", err)
+	}
+	docBytes, err := sb.Marshal()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal bundle: %w", err)
+	}
+
+	var o bytes.Buffer
+	if err := json.Indent(&o, docBytes, "", "    "); err != nil {
+		return nil, err
+	}
```

Comment:
I think my vote would be to leave indenting to the caller. Most callers won't care so it just ends up being wasted cycles.

Reply:
I went back and forth with this, and couldn't form a strong opinion. Finally thought that the content could be needed to be seen by a human at some point (where indenting is helpful) and returning it indented made sense to avoid the need of additional parsing from the consumer if that would happen in the majority of the cases.
But it's just a couple of lines, and as you are saying, if it's not needed it would be wasting cycles without reason.
So, it's good that you have a stronger opinion here :)
I'll remove the Indent call.